core/src/main/java/tc/oc/pgm/kits/KitParser.java

```diff
@@ -398,6 +405,53 @@ public ItemStack parseHead(Element el) throws InvalidXMLException {
     return itemStack;
   }
 
+  public ItemStack parseFirework(Element el) throws InvalidXMLException {
+    ItemStack itemStack = parseItem(el, Material.FIREWORK);
+    FireworkMeta meta = (FireworkMeta) itemStack.getItemMeta();
+    int power = XMLUtils.parseNumber(Node.fromAttr(el, "power"), Integer.class, false, 1);
+    meta.setPower(power);
+
+    for (Element explosionEl : el.getChildren("explosion")) {
+      Type type =
+          XMLUtils.parseEnum(Node.fromAttr(explosionEl, "type"), Type.class, null, Type.BURST);
+      boolean flicker = XMLUtils.parseBoolean(Node.fromAttr(explosionEl, "flicker"), false);
+      boolean trail = XMLUtils.parseBoolean(Node.fromAttr(explosionEl, "trail"), false);
+
+      List<Color> primary = parseColors(Node.fromChildren(explosionEl, "color"));
+      List<Color> fade = parseColors(Node.fromChildren(explosionEl, "fade"));
+
+      if (primary.isEmpty()) {
+        throw new InvalidXMLException("At least one <color> must be defined", explosionEl);
+      }
+
+      meta.addEffect(
+          FireworkEffect.builder()
+              .with(type)
+              .withColor(primary)
+              .withFade(fade)
+              .flicker(flicker)
+              .trail(trail)
+              .build());
+    }
+
+    itemStack.setItemMeta(meta);
+    return itemStack;
+  }
+
+  private List<Color> parseColors(List<Node> nodes) throws InvalidXMLException {
+    List<Color> colors = Lists.newArrayList();
+    for (Node node : nodes) {
+      String rawColor = node.getValue();
+      if (rawColor != null) {
+        if (!rawColor.matches("[a-fA-F0-9]{6}")) {
```

Comment:
Do we not have some util to parse these? Leather armor had to parse colors before this

Reply:
@Pablete1234 Not that I'm aware of. I looked into how leather armor has its color applied:

https://github.com/PGMDev/PGM/blob/59c1dc9f2719fd85165451bf502f921756b3a8e4/core/src/main/java/tc/oc/pgm/kits/KitParser.java#L514-L524

Would it be beneficial to move the color parsing logic to XMLUtils?  Perhaps something like
```java
public static Color parseHexColor(Node node, Color def) throws InvalidXMLException {
      String rawColor = node.getValue();
      if (rawColor != null) {
        if (!rawColor.matches("[a-fA-F0-9]{6}")) {
          throw new InvalidXMLException("Invalid color format", rawColor);
        }
        return Color.fromRGB(Integer.parseInt(rawColor, 16));
      }
      return def;
  }
  ```

Reply:
yeah, given at least 2 places are parsing these now, they should be moved over to an util for sure